### PR TITLE
Port search

### PR DESCRIPTION
**Pull request by nmcobb**
Updated to use global setup_vehicle. Created quadplane class

### CODE REVIEW

File: src/vtol.py [hunk 1]
```diff
@@ -184,9 +157,71 @@ def change_status(self, new_status):
...
+        '''
+        Note that from AC3.3 the message should be re-sent more often than every
+        second, as an ATTITUDE_TARGET order has a timeout of 1s.
+        In AC3.2.1 and earlier the specified attitude persists until it is canceled.
+        The code below should work on either version.
+        Sending the message multiple times is the recommended way.
+        '''
+        self.send_attitude_target(
+            roll_angle, pitch_angle, yaw_angle, yaw_rate, use_yaw_rate, thrust
+        )
```
Comment (Mr-Recalculate) [inline src/vtol.py]: does this block need to exist? I see that this same code is also in the loop below

File: src/vtol.py [hunk 2]
```diff
@@ -101,14 +74,13 @@ def start_auto_mission(self):
...
-        if self.configs["vehicle_type"] == "Quadcopter":
-            msg = self.message_factory.command_long_encode(
-                0, 0,    # target_system, target_component
-                mavutil.mavlink.MAV_CMD_MISSION_START, #command
-                0, #confirmation
-                0, 0, 0, 0, 0, 0, 0)    # param 1 ~ 7 not used
-            # send command to vehicle
-            self.send_mavlink(msg)
+        msg = self.message_factory.command_long_encode(
+            0, 0,    # target_system, target_component
```
Comment (Mr-Recalculate) [inline src/vtol.py]: why are these two parameters zero? I see you refer to them as target_system and target_component is there any point in which these parameters would not be zero? if so they should be assigned to variables.

File: src/vtol.py [hunk 3]
```diff
@@ -155,9 +127,10 @@ def go_to(self, point):
                 break
         print("Target reached")
 
+
     def land(self):
         '''Commands vehicle to land'''
-        self.mode = VehicleMode("LAND")
+        self.mode = VehicleMode(self.land_mode)
```
Comment (Mr-Recalculate) [inline src/vtol.py]: I like that this was moved to the init method

File: src/vtol.py [hunk 4]
```diff
@@ -184,9 +157,71 @@ def change_status(self, new_status):
...
+            yaw_angle=None,
+            yaw_rate=0.0,
+            use_yaw_rate=False,
+            thrust=0.5,
+    ):
+        '''
+        use_yaw_rate: the yaw can be controlled using yaw_angle OR yaw_rate.
+                    When one is used, the other is ignored by Ardupilot.
+        thrust: 0 <= thrust <= 1, as a fraction of maximum vertical thrust.
+                Note that as of Copter 3.5, thrust = 0.5 triggers a special case in
```
Comment (Mr-Recalculate) [inline src/vtol.py]: I don't know if this in in our control or not. But is using a number in the range of possible thrusts really the best practice? additionally, why is thrust hard coded to 0.5? and if thrust is assigned to a variable number in between 0 and 1 what range would the hover mode activate in?

File: src/main.py [hunk 5]
```diff
@@ -1,25 +1,22 @@
...
 def main(configs):
     '''Configure vtol and ready for mission'''
-    vehicle = setup_vehicle(configs)
-
+    vehicle = setup_vehicle(configs, VTOL)
     vehicle.takeoff()
 
-    vehicle.set_altitude(3)
+    # vehicle.send_ned_velocity(-5, 3, 0, 5)
+    vehicle.set_attitude(pitch_angle=20, duration=5)
```
Comment (Mr-Recalculate) [inline src/main.py]: should these be hard coded?

File: src/quadplane_main.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+from util import setup_vehicle
+from quadplane import QuadPlane
+
+def main(configs):
+    '''Configure vtol and ready for mission'''
+    vehicle = setup_vehicle(configs, QuadPlane)
+
+    vehicle.parameters['Q_GUIDED_MODE'] = 1
+    vehicle.takeoff()
+    vehicle.set_altitude(10)
```
Comment (Mr-Recalculate) [inline src/quadplane_main.py]: should this be hard coded?

File: src/vtol.py (hunk 1 above)
Comment (nmcobb) [reply src/vtol.py]: Yeah, this is code from ardupilot, not something I wrote, I noticed this as well, I'll do some testing to see what, if any difference this makes.

File: src/vtol.py (hunk 2 above)
Comment (nmcobb) [reply src/vtol.py]: Again, this is from arupilot, I'm not particularly concerned about these for now

File: src/vtol.py (hunk 3 above)
Comment (nmcobb) [reply src/vtol.py]: Yeah, had to abstract as quadplane uses "QLAND" and quad uses "LAND", didnt want to rewrite code

File: src/vtol.py (hunk 4 above)
Comment (nmcobb) [reply src/vtol.py]: This is also determined my ardupilot, the command just takes a thrust between 0 and 1. We can experiment with what values to use as we go i guess... I agree that this metric is not particularly informative. We would most likely use PID for this so it doesn't matter

File: src/main.py (hunk 5 above)
Comment (nmcobb) [reply src/main.py]: Was just using this for a testing file for now

File: src/quadplane_main.py (hunk 6 above)
Comment (nmcobb) [reply src/quadplane_main.py]: again testing file, I will move the parameter swap to an initialize method though

**Pull request merged**